### PR TITLE
Add service inventories

### PR DESCRIPTION
**Pull request by Ladas**

### CODE REVIEW

File: db/migrate/20190926144242_add_service_inventories_table.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class AddServiceInventoriesTable < ActiveRecord::Migration[5.2]
+  def change
+    create_table :service_inventories, :id => :bigserial, :force => :cascade do |t|
+      t.references :tenant, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
+      t.references :source, :index => false, :null => false, :foreign_key => {:on_delete => :cascade}
+
+      t.string :source_ref, :null => false
+      t.string :name
+      t.string :inventory_type
```
Comment (agrare) [inline db/migrate/20190926144242_add_service_inventories_table.rb]: What are some examples of inventory_type?  Wondering if this should be in extra

**Comment by agrare** [comment]
@Ladas I think you're missing the `ServiceInventory` model

**Comment by agrare** [comment]
We should probably have `service_inventory_tags`

**Comment by Ladas** [comment]
@Ladas ah yeah, the model escaped the git add

Not sure about tags, there are none in tower. Is catalog planning to store tags here?

File: db/migrate/20190926144242_add_service_inventories_table.rb (hunk 1 above)
Comment (Ladas) [reply db/migrate/20190926144242_add_service_inventories_table.rb]: hm, maybe? In the account I have it's all `inventory`, so not sure what the types are

**Comment by agrare** [comment]
Yeah we wouldn't collect tags from the source, but the user might still want to assign tags to an inventory.  I've heard assigning the "production" tag to an inventory to invoke different checks.

File: db/migrate/20190926144242_add_service_inventories_table.rb (hunk 1 above)
Comment (Ladas) [reply db/migrate/20190926144242_add_service_inventories_table.rb]: I say this would distinct e.g. AWS inventory, inventory as a script, etc.

Comment (agrare) [reply db/migrate/20190926144242_add_service_inventories_table.rb]: I'm not sure what that means :)

Comment (Ladas) [reply db/migrate/20190926144242_add_service_inventories_table.rb]: ah ok, so it's basically binary now, let me move it to extra

`Type: Identifies whether it is a standard inventory or a Smart Inventory.`

https://docs.ansible.com/ansible-tower/latest/html/userguide/inventories.html#smart-inventories

Comment (agrare) [reply db/migrate/20190926144242_add_service_inventories_table.rb]: :+1:

Comment (Ladas) [reply db/migrate/20190926144242_add_service_inventories_table.rb]: done

**Pull request merged**